assets/app/data/services/upgrade/upgrade-admin.factory.js

```diff
@@ -0,0 +1,33 @@
+(function() {
+
+    angular
+        .module('crowbarData.upgrade')
+        .factory('upgradeUpgradeAdminFactory', upgradeUpgradeAdminFactory);
+
+    upgradeUpgradeAdminFactory.$inject = ['$q', '$http'];
+    /* @ngInject */
+    function upgradeUpgradeAdminFactory($q, $http) {
+        return {
+            getAdminUpgrade: getAdminUpgrade,
+            getAdminUpgradeStatus: getAdminUpgradeStatus
+        };
+
+        function getAdminUpgrade() {
+            var requestOptions = {
+                method: 'POST',
+                url: '/api/upgrade7/admin-upgrade'
+            };
+
+            return $http(requestOptions);
+        }
+
+        function getAdminUpgradeStatus() {
```

Comment:
The whole API integration needs to be (re)checked. Both the URLs and the contents of requests and responses. I had to assume something to go forward.
Another thing is that if we are going to use one endpoint for multiple things, maybe there should be some global polling service which will check it and expose the cached contents to the page controllers...